logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/message/SendTextMessageUseCaseTest.kt

```diff
@@ -0,0 +1,246 @@
+package com.wire.kalium.logic.feature.message
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.client.ClientRepository
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.id.PlainId
+import com.wire.kalium.logic.data.message.MessageRepository
+import com.wire.kalium.logic.data.user.UserRepository
+import com.wire.kalium.logic.framework.TestUser
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.sync.SyncManager
+import io.mockative.Mock
+import io.mockative.Times
+import io.mockative.anything
+import io.mockative.given
+import io.mockative.mock
+import io.mockative.verify
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.runTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertIs
+
+class SendTextMessageUseCaseTest {
+
+    @Mock
+    private val messageRepository = mock(MessageRepository::class)
+
+    @Mock
+    private val userRepository = mock(UserRepository::class)
+
+    @Mock
+    private val clientRepository = mock(ClientRepository::class)
+
+    @Mock
+    private val syncManager = mock(SyncManager::class)
+
+    @Mock
+    private val messageSender: MessageSender = mock(MessageSender::class)
+
+    lateinit var sendTextMessageUseCase: SendTextMessageUseCase
+
+    @BeforeTest
+    fun setup() {
+        sendTextMessageUseCase = SendTextMessageUseCase(
+            messageRepository = messageRepository,
+            userRepository = userRepository,
+            clientRepository = clientRepository,
+            syncManager = syncManager,
+            messageSender = messageSender
+        )
+    }
+
+    @Test
+    fun givenRecipients_whenCreatingAnEnvelope_thenProteusClientShouldBeUsedToEncryptForEachClient() = runTest {
+        //given
+        given(syncManager)
```

Comment:
Configure the mock to generate stubs for unit methods and you can remove this:

Example:
```
private val conversationDAO = configure(mock(ConversationDAO::class)) { stubsUnitByDefault = true }
```